modules/programs/waybar.nix

```diff
@@ -215,7 +215,17 @@ in {
       '';
     };
 
-    systemd.enable = mkEnableOption "Waybar systemd integration";
+    systemd = {
+      enable = mkEnableOption "Waybar systemd integration";
+
+      target = mkOption {
+        type = str;
+        default = "sway-session.target";
```

Comment:
What are the downsides to setting `graphical-session.target` as the default?

AFAIK the systemd option in home-manager's Sway module only adds `systemctl --user start sway-session.target` which does not restart the `sway-session.target`.
When a user logs out or its session crashes then `sway-session.target` is still active (and not (re)started)

Reply:
Thanks for the comment, I looked into it further, and I posted a more detailed explanation. It can, and probably should be the default.